examples/mugen/retrieval/eval.py

```diff
@@ -22,33 +21,32 @@ def get_yaml_config():
             "Please pass 'config' to specify configuration yaml file for running VideoCLIP evaluation"
         )
     yaml_conf = OmegaConf.load(cli_conf.config)
-    return OmegaConf.to_container(yaml_conf)
+    conf = instantiate(yaml_conf)
+    return conf
 
 
 def evaluate():
     args = get_yaml_config()
-    datamodule_args = args["datamodule_args"]
-    lightningmodule_args = args["lightningmodule_args"]
-    dataset_args = args["dataset_args"]
-    videoclip_args = args["videoclip_args"]
-    evaluation_args = args["evaluation_args"]
 
-    dataset_args = MUGENDatasetArgs(get_audio=False, **dataset_args)
+    dataset_args: MUGENDatasetArgs = args.dataset_args
     datamodule = MUGENDataModule(
         dataset_args,
-        text_transform=BertTextTransform(**datamodule_args["bert_text_transform"]),
-        video_transform=VideoTransform(**datamodule_args["video_transform"]),
-        batch_size=datamodule_args["batch_size"],
-        num_workers=datamodule_args["num_workers"],
-        shuffle=datamodule_args["shuffle"],
+        text_transform=BertTextTransform(
+            **vars(args.datamodule_args.bert_text_transform)
+        ),
+        video_transform=VideoTransform(**vars(args.datamodule_args.video_transform)),
+        batch_size=args.datamodule_args.batch_size,
+        num_workers=args.datamodule_args.num_workers,
+        shuffle=args.datamodule_args.shuffle,
     )
 
-    model = VideoCLIPLightningModule(**lightningmodule_args, **videoclip_args)
-    model = model.load_from_checkpoint(evaluation_args["checkpoint_path"])
-
-    trainer = Trainer(
-        accelerator=evaluation_args["accelerator"], devices=evaluation_args["devices"]
+    model = VideoCLIPLightningModule.load_from_checkpoint(
+        args.checkpoint_path,
+        **vars(args.lightningmodule_args),
+        **vars(args.videoclip_args)
     )
+
+    trainer = Trainer(accelerator="auto", devices=1)
```

Comment:
Why using `auto` for inference? My understanding is that typically training is done on GPU but inference on CPU? I would also probably set `device` to be a parameter instead of hardcoding it --- unless there's a strong reason for `1`?

Reply:
I thought having the option of using gpu for evaluation would be convenient with the larger test set, but i'm not sure about the convention. I hardcoded `devices=1` because of the recommendation in the lightning docs (https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#testing)

Reply:
I am not 100% sure about how these convention (which makes sense to me) compare to our "typical" ways of doing things. I'd recommend you either write a post or seek others' feedback on this. 
Approved to unblock though.

Reply:
MUGEN authors used 1 gpu but it sounds like cpu is also a good option when batch size is small. So I'm thinking of making accelerator configurable, but leaving the default as "auto"